Fix/s3

cmccully opened the pull request:
Fixes for migrating the archive to all s3.

cmccully committed:
Fixed hard coding of fits exchange.

Commit:
Remove trailing slash from ARCHIVE_FRAME_URL, as it was causing us to double up on slashes when constructing the final url

cmccully committed:
Changed directory where code lives so it doesn't get overwritten by mounting /nres

cmccully committed:
Removed a slash in api-root to be consistent with the ingester lib.

cmccully committed:
Added more verbose logging for unexpected exceptions

cmccully committed:
Updated version of the ingester.

cmccully committed:
Removed opentsdb metrics that were causing issues. Updated message to look for version set instead of path to tell if we are listening to the post archive queue or pre.

cmccully committed:
Fixes to paths for the beat schedules.

cmccully committed:
Fixed path issue in making summary pdfs.

cmccully committed:
Fixes for end of night plots.

cmccully committed:
Fixes for metrics.

cmccully committed:
More fixes for getting the archive endpoint.

cmccully committed:
Merge branch 'fix/s3' of github.com:lcogt/nres-pipe into fix/s3

cmccully committed:
Typo fix.

cmccully committed:
More bugfixes in end of night reports.

cmccully committed:
Fixed nightly retracing to use the archive api to get headers instead of reading the headers from disk.

cmccully committed:
Removed unnecessary args in nightly tracing.

Commit:
Update ingester version to use new metrics tag. Update SQLAlchemy version for security fix

cmccully committed:
Increased the length of the filename value in the dbs so that we don't truncate

cmccully committed:
Added permissions fix to python.

cmccully committed:
Pinned an astroquery dependency.

cmccully committed:
Removed chown that seems to hang in docker.

cmccully committed:
Added initial frame rejection to the listener to hopefully fix the backlog of messages on the archived_fits queue.

cmccully committed:
added a nasty hack to get around the fact that all these messages have db1 in them.

cmccully committed:
Fix to l1pubdat logic to work on leap years. Path fixes in the jenkins file.

cmccully committed:
Made logging in the ingester more verbose.

cmccully committed:
Another path fix.

cmccully committed:
Typo fix.

cmccully committed:
Fix to the db logic in the listener when a checksum changes.

cmccully committed:
Fix to the db logic in the listener when a checksum changes.

cmccully committed:
Fix to the db logic in the listener when a checksum changes.

Commit:
Update lco-ingester dependency version

The lco-ingester library has been updated to version 2.1.14, which
itself contains an update to one of its dependencies,
opentsdb-python-metrics, which publishes metrics data on the standard
HTTP port 80 by default, rather than using a custom port number.

This behavior is also now configurable, rather than being hard-coded
into the library.

With this change, the LCO IT staff will be able to migrate the OpenTSDB
server to our newer hardware platform.

mgdaily committed:
Merge branch 'fix/queue_backup' into upgrade-ingester-library

mgdaily committed:
Pin version of astroquery.

When merging changes from the base branch, I accidentally removed the pinning on astroquery. This commit re-instates that.

mgdaily committed:
Merge pull request #41 from LCOGT/upgrade-ingester-library

Update lco-ingester dependency version

cmccully committed:
Fixes to refine traces.

cmccully committed:
Merge branch 'fix/queue_backup' of github.com:lcogt/nres-pipe into fix/queue_backup

cmccully committed:
Merge pull request #40 from LCOGT/fix/queue_backup

queue backlog

mgdaily commented on nrespipe/utils.py:
```diff
@@ -578,12 +603,17 @@ def get_mag_from_simbad(target_name : str):
 
```
It might pay to add some retry logic in here. Your call, but it's saved BANZAI a few times when the archive was having problems. Or when we got a ChunkedEncodingError from s3.

https://github.com/LCOGT/banzai/blob/485a13362448a768a41f92c1a56d860d8469064a/banzai/utils/fits_utils.py#L219

mgdaily commented on nrespipe/utils.py:
```diff
@@ -578,12 +603,17 @@ def get_mag_from_simbad(target_name : str):
-    response = requests.get(url, headers=settings.ARCHIVE_AUTH_TOKEN, stream=True).json()
+    response = requests.get(url, headers=settings.ARCHIVE_AUTH_TOKEN).json()
     with open(os.path.join(output_directory, response['filename']), 'wb') as f:
         f.write(requests.get(response['url']).content)
```
Add a `stream=True` flag to the `requests.get` call here since we're dealing with large files from s3.

mgdaily commented on nrespipe/utils.py:
```diff
@@ -536,20 +531,50 @@ def extract_signal_to_noise_from_pdf(pdf_reader: PdfFileReader, output_table: Ta
-                                                                         for filename in filenames]
-    raw_files = get_files_without_extensions_and_e00(glob(os.path.join(raw_directory, '*e00.fits*')), '.fits.fz')
-    processed_files = get_files_without_extensions_and_e00(glob(os.path.join(specproc_directory, '*.tar.gz')), '.tar.gz')
+def query_archive_api(site, dayobs, rlevel=0, obstype='TARGET', telid='igla'):
```
As a matter of fact, you could add retries to all of these external service calls.

Tenacity makes it easy!™

mgdaily commented on nrespipe/utils.py:
```diff
@@ -55,8 +57,8 @@ def get_md5(filepath):
-    record = dbs.get_processing_state(filename, checksum, db_address)
+def already_processed(filename, checksum, db_address):
+    record = dbs.get_processing_state(filename, db_address)
     return not record.processed or checksum != record.checksum
```
I'm slightly confused by this logic. An image is considered "already processed" if its db record's processed field is False or the incoming checksum isn't equal to the db's checksum? `need_to_process` seems like a better name for this function - as that's what it feels like this logic is determining. Or am I just utterly confused?

mgdaily reviewed (commented):
Mostly looks good, just some suggestions and one bit of confusion.

mgdaily committed:
Update lco-ingester version

This is to deal with NRES calibration images that do not have RAs and DECs defined in their headers.

mgdaily committed:
Update to use OCS Ingester version 2.2.5

This ingester version includes updates related to the deletion of the "old" LCO s3 bucket. No functional changes.
